networkx/algorithms/link_prediction.py

```diff
@@ -275,11 +276,17 @@ def common_neighbor_centrality(G, ebunch=None, alpha=0.8):
            Sci Rep 10, 364 (2020).
            https://doi.org/10.1038/s41598-019-57304-y
     """
-    shortest_path = nx.shortest_path(G)
+
+    spl = dict(nx.shortest_path_length(G))
+    inf = float("inf")
 
     def predict(u, v):
-        return alpha * len(list(nx.common_neighbors(G, u, v))) + (1 - alpha) * (
-            G.number_of_nodes() / (len(shortest_path[u][v]) - 1)
+        if u == v:
+            raise nx.NetworkXAlgorithmError("Self links are not supported")
```

Comment:
Do any of the measures in `link_prediction` support self-loops? If not, it might be nice to see if this check could be moved out to `_apply_prediction`. Let's worry about that in a follow-up PR though.

Reply:
I just had a quick look. It looks like all the other measures do support self loops. This is tested by the "test_equal_nodes" tests. Interestingly the Jaccard coefficient and Preferential attachment methods also support self loops although they don't have the "test_equal_nodes" test. I might add a follow up PR to clean up the link prediction tests.